DataStructures/Timeline.cs

```diff
@@ -0,0 +1,577 @@
+/*
+    Author: Lorenzo Lotti
+    Name: Timeline (DataStructures.Timeline<TValue>)
+    Type: Data structure (class)
+    Description: A collection of dates/times and values sorted by dates/times easy to query.
+    Usage:
+        this data structure can be used to represent an ordered series of dates or times with which to associate values.
+        An example is a chronology of events:
+            306: Constantine is the new emperor,
+            312: Battle of the Milvian Bridge,
+            313: Edict of Milan,
+            330: Constantine move the capital to Constantinople.
+*/
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace DataStructures
+{
+    /// <summary>
+    /// A collection of <see cref="DateTime"/> and <see cref="TValue"/> sorted by <see cref="DateTime"/> field.
+    /// </summary>
+    /// <typeparam name="TValue">Value associated with a <see cref="DateTime"/>.</typeparam>
+    public class Timeline<TValue> :
+        ICollection<(DateTime Time, TValue Value)>,
+        IEquatable<Timeline<TValue>>
+    {
+        private List<(DateTime Time, TValue Value)> timeline = new List<(DateTime Time, TValue Value)>();
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Timeline{TValue}"/> class.
+        /// </summary>
+        public Timeline()
+        {
+            // todo: improve performance and consider removing unnecessary methods
```

Comment:
Codacy found an issue: [Complete the task associated to this 'TODO' comment.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=5574777)